### PR TITLE
refactor(core): introduce the concept of Zone

### PR DESCRIPTION
**Pull request by ccamel**
## Purpose

This PR introduces the concept of `Zone` and revisit the concepts as required by #127.

### Zone (✏️)

The concept of `Zone` replaces the concept of `Dataspace` with a broader scope of application. The specification of this concept aligns with the common understanding that we share at OKP4.

```
Class: Zone    
    SubClassOf: 
        'has identifier' only xsd:anyURI,
        'has registrar' only 'Decentralized Identifier URI'
```

### Governance (➖)

The concept of governance is being removed. This concept was introduced to embody governance itself, independent of the `Dataspace` to which it applies. However, it has received some comments (i.e., https://github.com/okp4/ontology/pull/108#pullrequestreview-1376167565), particularly because governance was seen as a resource of the dataverse. Since the boundaries of the `Zone` concept are different and the way governance of a `Zone` is perceived has slightly changed, it seems appropriate to remove it.

This choice is open to discussion, but I believe it helps to make things simpler to understand without diminishing the range of possibilities.

### Resource (✏️)

The concept of `Resource` is evolving slightly. Firstly, by removing the property of belonging to the `Dataset`, which is no longer necessary. Secondly, by considering the concept not as a set union of other concepts in the dataverse (such as `Dataset` or `Service`), but as the common ancestor of these concepts in an "is-a" relationship. I believe this approach of modeling is more natural and probably easier to grasp and query.

```
Class: Resource
```

### Service  (✏️)

As a consequence of the previous point, a `Service` is now a subclass of `Resource`.

```
Class: Service
    SubClassOf: 
        Resource,
        'has identifier' only xsd:anyURI,
        'has identity' only 'Decentralized Identifier URI',
        'has registrar' only 'Decentralized Identifier URI'
```

### Dataset  (✏️)

As a consequence of the previous point, a `Dataset` is now a subclass of `Resource`.

```
Class: Dataset
    SubClassOf: 
        Resource,
        'has identifier' only xsd:anyURI,
        'has registrar' only 'Decentralized Identifier URI',
        'provided by' only Service
```

### ServiceExecutionOrder (✏️)

If `Resources` no longer have an explicit link with a `Zone` (property `belongsTo`), `ServiceExecutionOrders` must explicitly reference the `Zone` in which the orchestration should be executed. This is mainly because the `Zone` defines the economic mechanisms that apply.

```
Class: 'Service Execution Order'
    EquivalentTo: 
        ('belongs to' only Zone)
         and ('has contracting party' only 'Decentralized Identifier URI')
         and ('has contractor' only Service)
         and ('has object' only Service)
```

### zone:GeneralMetadata (✏️)

The namespace in which the `GeneralMetadata` profile describing the `Zone` has changed from <https://ontology.okp4.space/dataverse/dataspace/> to <https://ontology.okp4.space/metadata/zone/>.

## Tests

The example files have been updated accordingly, but there may be errors. :smile:

### CODE REVIEW

File: src/metadata-dataset-general.ttl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 @prefix owl: <http://www.w3.org/2002/07/owl#> .
 @prefix core: <https://ontology.okp4.space/core/> .
 @prefix license: <https://ontology.okp4.space/thesaurus/license/> .
-@prefix meta: <https://ontology.okp4.space/metadata/dataset/> .
+@prefix meta: <https://ontology.okp4.space/metadata/zone/> .
```
Comment (MdechampG) [inline src/metadata-dataset-general.ttl:6]: Is it normal to replace `dataset` by `zone` here?

**Review by MdechampG — COMMENTED** [review]
> 
> zone:GeneralMetadata (✏️)
> 
> The namespace in which the GeneralMetadata profile describing the Zone has changed from https://ontology.okp4.space/metadata/dataset/ to https://ontology.okp4.space/metadata/zone/.
> 

I don't understand why the profile "GeneralMetadata", which concerns datasets, is now in the namespace `/metadata/zone` 

Except from that, everything looks good, thank you 👍🏻

**Review by fredericvilcot — APPROVED** [review]
Great work 👌

**Comment by fredericvilcot** [comment]
@MdechampG 
Regarding this PR, we should create an issue to handle the changes in our front app, no?

**Review by CedrickArmel — APPROVED** [review]

**Review by amimart — APPROVED** [review]
Great!

I especially like the idea of the Governance being only a descriptive element of a Zone 😉

**Comment by ccamel** [comment]
> `Resources` no longer have an explicit link with a zone (belongsTo property), we have deleted this property do we need to keep the prefix `zone` : https://ontology.okp4.space/dataverse/zone/ in the dataset and services examples? Or will we change this in another PR when we update the examples?

Well, it's not very important, but yes, it would be clearer to remove prefixes that are no longer needed in examples. Would you like to do it? :stuck_out_tongue:

**Review by SonnevilleDamien — APPROVED** [review]

**Comment by MdechampG** [comment]
> @MdechampG Regarding this PR, we should create an issue to handle the changes in our front app, no?

@fredericvilcot Yes you're right we need to adapt the portal in accordance to these modifications.

**Review by MdechampG — APPROVED** [review]
Everything looks good thank you ! :rocket:

**Pull request merged**